index.js

```diff
@@ -34,18 +38,31 @@ EncryptedField.prototype.vault = function(name) {
 
             previous = new Buffer(previous);
 
-            var iv = previous.slice(0, self._iv_length);
-            var content = previous.slice(self._iv_length, previous.length);
-            var decipher = crypto.createDecipheriv(self._algorithm, self.key, iv);
+            function decrypt(key) {
+                var iv = previous.slice(0, self._iv_length);
+                var content = previous.slice(self._iv_length, previous.length);
+                var decipher = crypto.createDecipheriv(self._algorithm, key, iv);
 
-            var json = decipher.update(content, undefined, 'utf8') + decipher.final('utf8');
-            return JSON.parse(json);
+                var json = decipher.update(content, undefined, 'utf8') + decipher.final('utf8');
+                return JSON.parse(json);
+            }
+
+            var keyCount = self.decryptionKeys.length;
+            for (var i = 0; i < keyCount; i++) {
+                try {
+                    return decrypt(self.decryptionKeys[i]);
+                } catch (error) {
```

Comment:
This is definitely the thing that's most shaky about this. An error does indeed throw when using the default (`aes-256-cbc`), but its hard to tell how bound to the specific encryption method that is